### ISSUE TITLE
[QA] question about dataset label 

### ISSUE THREAD

**Issue by 1163307648** [InternLM/InternLM#318]
### Describe the question.

In the process of fine-tuning training using the Alpaca framework, I noticed that in the code, labels are shifted by one position relative to tokens when constructing the dataset. 
![image](https://github.com/InternLM/InternLM/assets/22743886/a743a939-c53b-45d3-9527-5e86708b372b)
Then, when calculating the loss in modeling_internlm.py, another shift of one position is applied to the labels. 
![image](https://github.com/InternLM/InternLM/assets/22743886/331461bc-bf10-4718-9af7-932d90d75f8b)
Is this an extra shift, or have I misunderstood it? Not familiar with NLP, forgive me.Thank you for your response.
